### ISSUE TITLE
Improve code coverage

### ISSUE THREAD

**Issue by bstrie** [enarx/enarx#1894]
As of this writing our overall code coverage sits at 58%: https://app.codecov.io/gh/enarx/enarx . It would be nice to get that somewhere north of 90%. People have mentioned that there are some difficulties related to our ability to instrument the shims which is dragging down our percentage, so if something that we're doing is truly untestable then we should try to find a way to exclude the untestable portions from the coverage results, because otherwise it's easy to miss when we're not testing something that we should be testing.

**Comment by haraldh** [comment]
|                   |        |
|-------------------|-------:|
| src               | 82.08% |
| exec-wasmtime | 57.94% |
| sallyport     | 84.16% |
| shim-kvm      |  4.56% |
| shim-sgx      | 14.13% |

**Comment by haraldh** [comment]
With #1899 I marked untestable regions... other functions/methods could be tested with mocked objects in `#[test]` functions.

**Comment by haraldh** [comment]
@jarkkojs could mark those regions as well for the `shim-sgx`.

**Comment by jarkkojs** [comment]
> @jarkkojs could mark those regions as well for the `shim-sgx`.

`#![cfg_attr(coverage, no_coverage)]` to the beginning of the file?

**Comment by haraldh** [comment]
no, per function.   see https://github.com/enarx/enarx/pull/1901

**Comment by bstrie** [comment]
@jarkkojs keeping in mind that we only want to use that approach on functions that *cannot* be tested, and not just functions that we don't currently test.
